### PR TITLE
[SPARK-17207][MLLIB]fix comparing Vector bug in TestingUtils

### PR DESCRIPTION
**Pull request by mpjlu**
## What changes were proposed in this pull request?

fix comparing Vector bug in TestingUtils.
There is the same bug for Matrix comparing. How to check the length of Matrix should be discussed first.  
## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala [hunk 1]
```diff
@@ -154,7 +154,11 @@ object TestingUtils {
      */
     def absTol(eps: Double): CompareVectorRightSide = CompareVectorRightSide(
       (x: Vector, y: Vector, eps: Double) => {
-        x.toArray.zip(y.toArray).forall(x => x._1 ~= x._2 absTol eps)
+        if (x.size != y.size) {
```
Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala]: How about just `x.size == y.size && ...`?
I think this also needs to be done for the matrix ops below.

Comment (mpjlu) [inline mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala]: Yes, thanks. 
How about check matrix length by
x.numRows == y.numRows && x.numCols == y.numCols && ... ?

Comment (srowen) [inline mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala]: Agree with that, yes.

Comment (mpjlu) [inline mllib/src/test/scala/org/apache/spark/mllib/util/TestingUtils.scala]: code is updated. Thanks @srowen

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Jenkins add to whitelist

**Comment by dbtsai** [comment]
Can you also fix `https://github.com/apache/spark/blob/master/mllib-local/src/test/scala/org/apache/spark/ml/util/TestingUtils.scala`? Please add tests showing the issue is addressed. Thanks.

**Comment by mpjlu** [comment]
Sure, I will fix it, and add test cases. thanks. @dbtsai ,

**Comment by mpjlu** [comment]
Hi @dbtsai , PR 2294 added Matrix comparing in TestingUtils, but did not add any test cases in TestingUtilsSuite. I did not add test cases for Matrix comparing in the PR either.
If Matrix comparing cases are required. I can add all test cases them, not only test this PR, like Vector comparing test cases.

**Comment by dbtsai** [comment]
Please also add test cases for matrices. Thanks.

**Comment by dbtsai** [comment]
LGTM. Merged into master. Thanks.

**Pull request closed without merging**